src/Slider/Slider.js

```diff
@@ -37,51 +37,128 @@ const valueInRangePropType = (props, propName, componentName) => {
   }
 };
 
+const crossAxisProperty = {
+  x: 'height',
+  'x-reverse': 'height',
+  y: 'width',
+  'y-reverse': 'width',
+};
+
+const crossAxisOffsetProperty = {
+  x: 'top',
+  'x-reverse': 'top',
+  y: 'left',
+  'y-reverse': 'left',
+};
+
+const mainAxisProperty = {
+  x: 'width',
+  'x-reverse': 'width',
+  y: 'height',
+  'y-reverse': 'height',
+};
+
+const mainAxisMarginFromEnd = {
+  x: 'marginRight',
+  'x-reverse': 'marginLeft',
+  y: 'marginTop',
+  'y-reverse': 'marginBottom',
+};
+
+const mainAxisMarginFromStart = {
+  x: 'marginLeft',
+  'x-reverse': 'marginRight',
+  y: 'marginBottom',
+  'y-reverse': 'marginTop',
+};
+
+const mainAxisOffsetProperty = {
+  x: 'left',
+  'x-reverse': 'right',
+  y: 'bottom',
+  'y-reverse': 'top',
+};
+
+const mainAxisClientProperty = {
+  x: 'clientWidth',
+  'x-reverse': 'clientWidth',
+  y: 'clientHeight',
+  'y-reverse': 'clientHeight',
+};
+
+const mainAxisClientOffsetProperty = {
+  x: 'clientX',
+  'x-reverse': 'clientX',
+  y: 'clientY',
+  'y-reverse': 'clientY',
+};
+
+const reverseMainAxisOffsetProperty = {
+  x: 'right',
+  'x-reverse': 'left',
+  y: 'top',
+  'y-reverse': 'bottom',
+};
+
+const isMouseControlInverted = (axis) => axis === 'x-reverse' || axis === 'y';
+
 const getStyles = (props, context, state) => {
   const {slider} = context.muiTheme;
   const fillGutter = slider.handleSize / 2;
   const disabledGutter = slider.trackSize + slider.handleSizeDisabled / 2;
   const calcDisabledSpacing = props.disabled ? ` - ${disabledGutter}px` : '';
+  const axis = props.axis;
 
   const styles = {
     slider: {
       touchCallout: 'none',
       userSelect: 'none',
       cursor: 'default',
-      height: slider.handleSizeActive,
+      [crossAxisProperty[axis]]: slider.handleSizeActive,
+      [mainAxisProperty[axis]]: '100%',
```

Comment:
Well, a div's default width is 100% -- default height is auto (based on content) -- but that doesn't work for this. Maybe there's a better calculation than `height: 100%`? Not sure.